### PR TITLE
146: notification animations

### PR DESCRIPTION
**Pull request by lucasrasec**
Implementing fade animations in notification component. 
- It was created  a file on utils to contain the classes of animations, so they have the possibility to be used in other components as well.

### CODE REVIEW

File: projects/ion/src/lib/notification/notification.component.ts [hunk 1]
```diff
@@ -53,7 +58,10 @@ export class NotificationComponent {
...
 
   closeNotification() {
-    this.notification.nativeElement.remove();
+    this.notification.nativeElement.classList.add(this.fadeOut);
+    setTimeout(() => {
+      this.notification.nativeElement.remove();
+    }, 1000);
   }
 
   closeAuto(closeIn: number = this.timeByWords(this.message)) {
```
Comment (imrochamatheus) [inline projects/ion/src/lib/notification/notification.component.ts:67]: Very cool idea of ​​timeByWords

**Review by imrochamatheus — APPROVED** [review]

File: projects/ion/src/lib/notification/notification.component.ts [hunk 2]
```diff
@@ -53,7 +58,10 @@ export class NotificationComponent {
   }
 
   closeNotification() {
-    this.notification.nativeElement.remove();
+    this.notification.nativeElement.classList.add(this.fadeOut);
+    setTimeout(() => {
+      this.notification.nativeElement.remove();
+    }, 1000);
```
Comment (iurynogueira) [inline projects/ion/src/lib/notification/notification.component.ts:64]: Why this timeout?

File: projects/ion/src/lib/utils/fadeAnimations.scss [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+  transform: none;
+  }
+} 
+.fadeInUp {
+  -webkit-animation-name: fadeInUp;
+  animation-name: fadeInUp;
+  -webkit-animation-duration: 1s;
+  animation-duration: 1s;
+  -webkit-animation-fill-mode: both;
+  animation-fill-mode: both;
```
Comment (iurynogueira) [inline projects/ion/src/lib/utils/fadeAnimations.scss:89]: You can use mixin for don't repeat this code in all class

**Review by iurynogueira — CHANGES_REQUESTED** [review]
Nice work!

File: projects/ion/src/lib/notification/notification.component.ts (hunk 2 above)
Comment (lucasrasec) [reply projects/ion/src/lib/notification/notification.component.ts:64]: This way the element will only be removed when the animation is complete. Without this TimeOut the "remove" action just destroys the component before the animation, interrupting the animation and sudenly disapearing with the element in the View

Comment (mizek1) [reply projects/ion/src/lib/notification/notification.component.ts:64]: setTimeout can be replaced by delay operator of RxJS

**Comment by iurynogueira** [comment]
add delay please 😃

File: projects/ion/src/lib/utils/fadeAnimations.scss [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+@-webkit-keyframes fadeOutUp {
+  0% {
+  opacity: 1;
+  }
+  100% {
+  opacity: 0;
+  -webkit-transform: translate3d(0, -100%, 0);
+  transform: translate3d(0, -100%, 0);
+  }
+}
```
Comment (iurynogueira) [inline projects/ion/src/lib/utils/fadeAnimations.scss:228]: You can add conditional to add mixin here, we can informe the position and change parameters 100% in correct position using this https://sass-lang.com/documentation/at-rules/control/if

**Review by iurynogueira — CHANGES_REQUESTED** [review]

**Review by Deeved — APPROVED** [review]

**Review by iurynogueira — APPROVED** [review]

**Pull request merged**